substrate/codec/src/codec.rs

```diff
@@ -49,7 +49,10 @@ impl<'a> Input for &'a [u8] {
 #[cfg(feature = "std")]
 impl<R: ::std::io::Read> Input for R {
 	fn read(&mut self, into: &mut [u8]) -> usize {
-		(self as &mut ::std::io::Read).read(into).unwrap_or(0)
```

Comment:
Apparently reading from a file with `read` is interrupted on the buffer boundary.